File: Darts/src/code/bean/ComboBoxGameType.java
package code.bean;

import javax.swing.DefaultComboBoxModel;
import javax.swing.JComboBox;

import bean.ComboBoxItem;
import code.db.GameEntity;
import object.HandyArrayList;

public class ComboBoxGameType extends JComboBox<ComboBoxItem<Integer>>
{
	public ComboBoxGameType()
	{
		DefaultComboBoxModel<ComboBoxItem<Integer>> model = new DefaultComboBoxModel<>();
		
		HandyArrayList<Integer> gameTypes = GameEntity.getAllGameTypes();
		for (int gameType : gameTypes)
		{
			ComboBoxItem<Integer> item = new ComboBoxItem<>(gameType, GameEntity.getTypeDesc(gameType));
			model.addElement(item);
		}
		
		setModel(model);
	}
	
	public int getGameType()
	{
		int ix = getSelectedIndex();
		ComboBoxItem<Integer> selectedItem = getItemAt(ix);
		return selectedItem.getHiddenData();
	}
}

File: src/main/kotlin/dartzee/bean/ComboBoxGameType.kt
package dartzee.bean

import dartzee.core.bean.ComboBoxItem
import dartzee.core.bean.selectedItemTyped
import dartzee.db.GameType
import dartzee.utils.getTypeDesc
import javax.swing.DefaultComboBoxModel
import javax.swing.JComboBox

class ComboBoxGameType : JComboBox<ComboBoxItem<GameType>>()
{
    init
    {
        val model = DefaultComboBoxModel<ComboBoxItem<GameType>>()

        val gameTypes = GameType.values()
        for (gameType in gameTypes)
        {
            val item = ComboBoxItem(gameType, getTypeDesc(gameType))
            model.addElement(item)
        }

        setModel(model)
    }

    fun getGameType() = selectedItemTyped().hiddenData
}
